src/components/Asset/AssetList.tsx

```diff
@@ -110,3 +132,37 @@ export const AssetList = <T extends Token>({
     </div>
   )
 }
+
+const AssetBalanceAdapter = ({
+  balances,
+  decimals,
+  defuseAssetId,
+}: {
+  balances: Record<string, bigint>
+  decimals: number
+  defuseAssetId?: string
+}) => {
+  const calculateBalance = useCallback(() => {
+    if (!balances || !defuseAssetId) {
+      return "0"
+    }
+    const tokenIds = defuseAssetId.split(",")
+    const isGroupedToken = tokenIds.length > 1
+    if (isGroupedToken) {
+      return tokenIds.reduce((acc, assetId) => {
+        const balance = formatUnits(balances[assetId] ?? BigInt(0), decimals)
+        return (Number(acc) + Number(balance)).toString()
+      }, "0")
+    }
+    if (defuseAssetId === "nep141:wrap.near") {
+      return formatUnits(
+        (balances[defuseAssetId] ?? BigInt(0)) +
+          (balances["near:native"] ?? BigInt(0)),
```

Comment:
Instead of storing not real defuse token (`near:native`) in balances, might be better to store a balance in separate variable. In that way we will be 100% sure that `defuseAssetId` is always token id within defuse protocol.